File: Client/scada-client/src/pages/AlarmInstance.js
import React from 'react';
import { Card, CardContent, Typography } from '@mui/material';

const AlarmInstance = ({ timestamp, message,value,criticalValue,units,alarmType}) => {
    const capitalizedAlarmType = alarmType.charAt(0).toUpperCase() + alarmType.slice(1).toLowerCase();

  return (
    <Card style={{ width: '99%', minHeight: '80px', marginBottom: '16px', borderRadius: '8px', border: '1px solid #ccc'}}>
      <CardContent>
        <Typography variant="subtitle2" color="textSecondary">
          {timestamp}
        </Typography>
        <Typography variant="body1">
          {message}
        </Typography>
        <Typography variant="body1">
          Alarm cutoff: {capitalizedAlarmType}er than {criticalValue}{units} 
        </Typography>
        <Typography variant="body1">
          Current Value: {value}{units}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default AlarmInstance;

File: SKADA/Models/Alarms/Repository/AlarmInstance.cs
using static SKADA.Models.Alarms.Model.Alarm;

namespace SKADA.Models.Alarms.Model
{
    public class AlarmInstance
    {
        public Guid Id { get; set; }
        public Guid AlarmId { get; set; }
        public DateTime Timestamp { get; set; }
        public string Message { get; set; }
        public string AlarmType { get; set; }
        public string Units { get; set; }
        public Double CriticalValue { get; set; }
        public Double Value { get; set; }

        public AlarmInstance()
        {

        }

        public AlarmInstance(Guid id, Guid alarmId, DateTime timestamp, string message, string alarmType, string units, double criticalValue, double value)
        {
            Id = id;
            AlarmId = alarmId;
            Timestamp = timestamp;
            Message = message;
            AlarmType = alarmType;
            Units = units;
            CriticalValue = criticalValue;
            Value = value;
        }
    }
}
